### DISCUSSION TITLE
[Issue]Copilot with VS 2022 is often very disruptive by cancelling out regular intellisense

### DISCUSSION THREAD

**Discussion by Cpt-Falcon** [community/community#15649] [Copilot Conversations] [answered]
I like copilot a lot and enjoy using it. However, it does not work with regular intellisense at all. I try to start typing the property and as I'm about to hit enter to auto-insert the property name, copilot decides to shutdown intellisense and make some suggestion that isn't correct. So I have to type and hit enter as fast as possible so that copilot doesn't have time to analyze the text and come up with a recommendation.

Instead of deleting the intellisense suggestion, copilot should somehow enable both options. Maybe something like: I can click the down arrow to select the intellisense suggestion. EIther way its a major nuisance that copilot essentially disables intellisense for all intents and purposes. I just don't want copilot to interrupt me when I'm about to select intellisense or give me the ability to choose the copilot or intellisense recommendation. I'd be happy to provide a video if necessary. In fact, why even hide the intellisense suggestion? I should be able to hit enter to accept the intellisense recommendation or tab to accept the copilot recommendation.

**Comment by CePur** [comment]
It is biggest issue for me while using the extension. Thinking to uninstall until this fixed somehow. Really frustrating.

Should show both intellisense and copilot and have ability to pick one.

**Reply by MarkBrainfire** [reply]
Agreed.  I'm cancelling my subscription.

**Reply by ayim** [reply]
Aaron from VS Copilot team here - we're aware of this and looking into it.
-A

**Reply by dpierson0721** [reply]
Have you reached something?

**Reply by ayim** [reply]
Yes - launched this feature in June.
Got some additional ideas around using IntelliSense to steer the model in the works.

https://devblogs.microsoft.com/visualstudio/github-copilot-visual-studio-intellisense/

**Comment by Pretagonist** [comment]
Have had this issue multiple time. 

Write variable name, press .  intellisense pops up sane menu, press tab for required property, copilot fills my code with garbage.

**Reply by ayim** [reply, marked as answer]
Absolutely! IntelliSense working together with Copilot's increased quality of the predictions as well since the list limits the model to valid methods and parameters.

We've had this feature around since late June so if it's not working for you, try reinstalling the latest version of the extension from the marketplace (https://marketplace.visualstudio.com/items?itemName=GitHub.copilotvs).

Demo: https://devblogs.microsoft.com/visualstudio/github-copilot-visual-studio-intellisense/

**Reply by arnonax-tr** [reply]
I just installed it, and it feels a bit better (I can't tell exactly why), but it's still get suggestions that are not aligned with IntelliSense, and if I press TAB (which is what I'm used to when using IntelliSense) I still get CoPilot's completion instead of IntelliSense. Here's an example:
![image](https://github.com/community/community/assets/62980738/e2ba5fc2-cb1c-47c5-bc42-c7dd7db67cfa)

**Reply by ayim** [reply]
@arnonax-tr this doesn't look like my IntelliSense, are you using Resharper?

**Reply by arnonax-tr** [reply]
Yes I do

**Comment by Cpt-Falcon** [comment]
Just thought I would give everyone a heads up if you're using resharper. I found a great solution. There is a check box that is automatically enabled for Disable ReSharper automatic competition while co pilot suggestion is shown:

![image](https://user-images.githubusercontent.com/10540204/167877507-0486ff3a-dc4c-4cc0-96eb-10726fa5bd20.png)

When you uncheck this box it works amazingly well. You can either select the resharper intellisense or tab to select github copilot. It seriously is a lifesaver for using both.

**Reply by tvbic** [reply]
it works perfectly

**Reply by mike-rowley** [reply]
I've been searching for this solution!  Thank You!

**Reply by artizzan** [reply]
Thank you for this! I completely missed that option!

**Reply by 03c** [reply]
Found this solution within 2 mins of searching. Why have I been suffering for months with this issue! 🤷

**Reply by TheDevelolper** [reply]
Honestly, your post saved me hours! Thank you so much. It works perfectly!

**Reply by nadinator** [reply]
Does this also work with python?

**Reply by asheroto** [reply]
Is there a way to do this if you don't use ReSharper?

**Reply by ayim** [reply]
Absolutely! IntelliSense working together with Copilot's increased quality of the predictions as well since the list limits the model to valid methods and parameters.
https://devblogs.microsoft.com/visualstudio/github-copilot-visual-studio-intellisense/

**Reply by hfloyd** [reply]
Unfortunately, it seems that this is no longer an option in later versions of ReSharper...

**Reply by shoff** [reply]
> Unfortunately, it seems that this is no longer an option in later versions of ReSharper...

Did you ever find a similar setting?

**Reply by hfloyd** [reply]
Nope

**Comment by BieleckiLtd** [comment]
That's my remark too. It would be nice if IntelliSense/IntelliCode would not be overridden by Copilot. They are complementary services.

Linked issues #9650, #9430, #11156, #14157, #11410

**Reply by ayim** [reply]
Absolutely! IntelliSense working together with Copilot's increased quality of the predictions as well since the list limits the model to valid methods and parameters.
https://devblogs.microsoft.com/visualstudio/github-copilot-visual-studio-intellisense/

**Reply by hfloyd** [reply]
Sorry, @ayim CoPilot overriding standard Intellisense by default is still an issue in even the most recent versions. Also, I've found that previously useful things  like "///" to automatically create a summary documentation template block just don't work any longer.

**Reply by ayim** [reply]
@hfloyd which language are you writing in? C++ and C# should have this support unless you're using Resharper

**Reply by hfloyd** [reply]
Hi @ayim I do have ReSharper installed. I am using C#.

**Reply by ayim** [reply]
@hfloyd I'm tracking this in a feedback item here:
https://developercommunity.visualstudio.com/t/GitHub-Copilot-Completions-in-Visual-Stu/10688821

Curious ... on a scale of 1-10, how critical is Resharper's IntelliSense to your overall use of Resharper?
If you were to use Reshaper but with Visual Studio's IntelliSense, how unsatisfied would you be on a scale of 1-10?

![image](https://github.com/user-attachments/assets/97bf38f8-a8de-4ab7-b317-72dab416b6b0)


--
Aaron Yim
Product Manager for GitHub Copilot in Visual Studio
[aayim@microsoft.com](mailto:aayim@microsoft.com)

**Reply by hfloyd** [reply]
I tested turning off the ReSharper IntelliSense, and didn't love the alternative. It just seems to be missing things, like if I type in all lowercase, RS would still pull up matching properties/methods (case insensitive), but the built-in wouldn't. 

After turning RS back on, I did notice that the "///" summary template was working again (so perhaps a VS update did include that improvement). I assume that these niggling issues are being addressed, and I will continue to post anything specific I find to be missing or suboptimal.

**Comment by sur1969** [comment]
Installed the VS2022 extension today and after an hour of use I've disabled it. Worked well in Visual Studio Code.

**Reply by Danyalkasiri** [reply]
****

**Comment by MulleDK19** [comment]
I have the opposite issue. Copilot never suggests anything if Intellisense is enabled. I constantly have to hit escape to cancel Intellisense and allow Copilot to suggest.

**Reply by ayim** [reply]
Yes - the issue prior to us launching the fix was you would have to choose between IntelliSense and Copilot (could not have both up at the same time).

Since June our telemetry has noticed an improvement in accept rates from both working together (since the list of valid methods in IntelliSense also helps reduce invalid code suggestions)

https://devblogs.microsoft.com/visualstudio/github-copilot-visual-studio-intellisense/

**Comment by endasil** [comment]
I also find this extremely frustrating.. Copilot is not useful with visual studio right now since it just kills the IntelliSense suggestions and gives me a new row instead.

**Reply by ayim** [reply]
I wonder if your Copilot extension was having trouble autoupdating ... we first launched this functionality in late June.

Try installing the most recent VSIX from this page: https://marketplace.visualstudio.com/items?itemName=GitHub.copilotvs

https://devblogs.microsoft.com/visualstudio/github-copilot-visual-studio-intellisense/

**Comment by aleksa-codes** [comment]
Glad I am not the only one, really would like a fix for this, my Copilot suggestion doesn't even show because intellisense is interrupting it.

**Reply by ayim** [reply]
I wonder if your Copilot extension was having trouble autoupdating ... this functionality should've been available for a couple months at the time of writing.

Try installing the most recent VSIX from this page: https://marketplace.visualstudio.com/items?itemName=GitHub.copilotvs

https://devblogs.microsoft.com/visualstudio/github-copilot-visual-studio-intellisense/

**Comment by AppFzx** [comment]
Same. I'm turning it off and cancelling before I start getting charged. This is not ready for market.

**Reply by ayim** [reply]
I wonder if your Copilot extension was having trouble autoupdating ... this functionality should've been available for a couple months at the time of writing (launched in June)

Try installing the most recent VSIX from this page: https://marketplace.visualstudio.com/items?itemName=GitHub.copilotvs

https://devblogs.microsoft.com/visualstudio/github-copilot-visual-studio-intellisense/

**Comment by mpottinger** [comment]
This is also an issue with copilot in other IDE's. Same problem in Pycharm. It often breaks intellisense, which is a bit frustrating.

**Comment by goleary** [comment]
completely defeats the purpose of having copilot if i can't use intellisense and copilot suggests incorrect code.

**Reply by Bokkeman** [reply]
The incorrect code is the worst aspect - it suggests code that references non-existent properties ... you'd think this was a really basic requirement.

**Reply by ayim** [reply]
@Bokkeman  @goleary absolutely - since launching this feature we've seen accept rates increase since IntelliSense helps inform Copilot of valid methods and parameters. Demo here: https://devblogs.microsoft.com/visualstudio/github-copilot-visual-studio-intellisense/

Please make sure you're on the latest version (1.139 at time of writing).
Consider reinstalling the extension with the latest version from the Marketplace; https://marketplace.visualstudio.com/items?itemName=GitHub.copilotvs

**Comment by cceyda** [comment]
There seems to be a solution for IntelliJ's but not VS code 😕 Having both is a must  https://github.com/orgs/community/discussions/14286#discussioncomment-2522711

**Reply by zenerbufen** [reply]
This discussion is about VS 2022, not VSCODE. You are off topic. 

It is frustrating, because EVERY thread about the issues with Visual Studio (NOT CODE) 2022 copilot issues is loaded with people giving suggestions, workarounds, claiming it is fixed already, or going off on tangents about other issues, all for VS CODE. There are plenty of VS CODE threads already, can we have a single thread about this for Visual Studio which isn't taken over by the CODE folks saying the issue doesn't exist so that we can get some attention on this from Microsoft / GitHub?

**Reply by ayim** [reply]
Hey folks - this functionality was actually available since June.
Demo here: https://devblogs.microsoft.com/visualstudio/github-copilot-visual-studio-intellisense/

Please make sure you're on the latest version (1.139 at time of writing).
Consider reinstalling the extension with the latest version from the Marketplace; https://marketplace.visualstudio.com/items?itemName=GitHub.copilotvs

**Comment by Voxar** [comment]
I solved it like this. 

Turn off "Automatically show inline completions"
<img width="421" alt="image" src="https://github.com/community/community/assets/45087/eeb8c50f-3ca0-41b5-8a32-40e5c83035b5">

Add keymap for "Trigger Inline Suggestion".
<img width="751" alt="image" src="https://github.com/community/community/assets/45087/5a586a00-4a9e-4a15-b41d-520704b3f388">

Normal suggestions pop up automatically and you get copilot suggestions only when you want them.

**Reply by CePur** [reply]
Is this for VS or VS Code ?

**Reply by TheDevelolper** [reply]
> Is this for VS or VS Code ?

No it's visual studio 2022

**Reply by CePur** [reply]
@TheDevelolper 


    No it's visual studio 2022

Where are these settings in Visual Studio 2022 ?

**Reply by zenerbufen** [reply]
These settings are not available in Visual Studio 2022. The poster is confused and is using Visual Studio Code instead, a separate software product with a TERRIBLE name that is causing untold amounts of confusion and bad reports on these forums as most people can't seem to tell them apart. 

![image](https://github.com/community/community/assets/1755919/259a57c8-b250-47f3-9d04-cc2e59f2f45e)

![image](https://github.com/community/community/assets/1755919/20579bd2-c205-4cde-b85c-dafb56025742)

As you can clearly see, the VS 2022 interface looks completely different from those provided as a solution. (because they are really from CODE)

the options "show next inline suggestion"
and 
"show inline" and "automatically show inline suggestion"
are not available options, only to completely disable co pilot globally.
Also missing from Visual Studio (NOT CODE) 2022 is the accept one word of suggestion keybind that code and other ide extensions has, making it difficult to take the part og copilot that is right, and ignore the stuff that is wrong.

**Reply by chulomanK** [reply]
Is this not solved yet for vs 2022?

**Reply by ayim** [reply]
Folks, Copilot autocomplete PM from VS2022 here.

1) This absolutely works with IntelliSense. In fact, we launched this feature first and have more IntelliSense based features we're working on that further improve quality (imagine if we got the number and order of method parameters correct almost all the time).

https://devblogs.microsoft.com/visualstudio/github-copilot-visual-studio-intellisense/

2) We found the "show next inline suggestion" shortcut usage really low; you're correct in the sense that this isn't available anymore.

**Comment by AleksiEPHVA** [comment]
The problems with this product are many. 

Copilot Chat:

UI:
- Unable to ctrl+c copy text from the chat window
- Pasting code into the chat window breaks if there are //comment tags in the code, also does not properly detect what is code and what is a question.

General:
- Copilot chat has the memory of a goldfish, does not recall things discussed two posts ago and have to repeat myself everytime I ask a new question.
- Flat out refuses to help with a function that checks the validity of Social security numbers.
- Does not explain why it refuses to answer a question or help with a subject.
- Does not give any context to the answer it has given
- Gives flat-out wrong code and refuses to do what I ask it to do (For example. Translate these words into english and add them to this class as public strings = it gives the same words as I gave it without translation)
- Buggy, on some answers I get a error window a.txt is not a valid filename fe.
- Has a low text characther amount in questions and also answers.
- filename"#" commands do not work properly, it wont read the selection properly.
- Does not have a  # workspace command like other products.
- Refuses to read cshtml files with the #file command. (Says something about copyright and about sensitive information)

Copilot auto code completion:

- Breaks intellisense as discussed above.
- Gives made up functions and class properties which do not exist.
- Need manual button binding to make use of it.
- Pressing tab does not take the suggestion but breaks the suggestion.

I have not tried VSCode version of this, only Visual studio 2022. Its a shame since I had high hopes for this after using chatGPT, which in my opinion is a far superior product and does not cost 19$ per seat per month.

**Reply by ayim** [reply]
@AleksiEPHVA which languages is GitHub Copilot breaking IntelliSense for?
We added support for C# and C++ that is expected to work as such: https://devblogs.microsoft.com/visualstudio/github-copilot-visual-studio-intellisense/

@rheapatel23's the expert on Chat here, would love to know what she thinks too

**Reply by asheroto** [reply]
Tested on VB.net and it's causing issues. I posted a video on one of these threads. You can search the discussions and find similar issues.

**Reply by ayim** [reply]
Gotcha - you're right. VB's one of the languages that needs some love when it comes to IntelliSense interaction

**Reply by rheapatel23** [reply]
Hi @asheroto, thank you for your feedback, I would love to learn more about this, would you be available for a call? 

Also I want to share if you turn on a preview feature using Tools-> Options...-> Search for "GitHub Copilot Chat", then select #Solution for C#. This will allow you to refer to your project :)

**Reply by AleksiEPHVA** [reply]
Im using c#

**Reply by AleksiEPHVA** [reply]
the # solution command does nothing it seems. It will not find references to functions or anything, what is it supposed to be doing?

**Reply by asheroto** [reply]
Hey @rheapatel23. Sure, I'm using VB.net, should I still use that preview feature you mentioned?

**Reply by ayim** [reply]
@asheroto AFAIK #solution is current for C# only, it wouldn't apply for VB .NET scenarios

**Reply by asheroto** [reply]
@ayim Ah, understood. Thanks

**Comment by adampaquette** [comment]
If you're like me and lost a lot of time due to the plugin suggesting copilot code instead of intellisence suggestions, uninstall the plugin then reinstall and it will be good.

**Reply by asheroto** [reply]
That does fix some things, but there are still issues. I've even installed VS 2022 on a new computer and had the same bugs.

**Reply by Skyppid** [reply]
Yeah I just uninstalled it completely. Not using it often enough and quality is still not good enough for most cases to accept the solution without more work to fix it afterwards. Had so many annoying issues with Copilot and IntelliSense not getting along, can't work like that. I stick with IntelliSense, it's worth more than some AI.

**Reply by ayim** [reply]
@Skyppid which language are you writing code in?

Copilot Completions <> IntelliSense should be working together in C++ and C#.
https://devblogs.microsoft.com/visualstudio/github-copilot-visual-studio-intellisense/

**Reply by Skyppid** [reply]
@ayim C#. I also have ReSharper running which has it's own IntelliSense - so might be a problem with that. Anyways, even in Rider I've seen that IntelliSense and AI tools don't work well together. I just keep using my own skills for now and try again in some time.

**Reply by ayim** [reply]
That explains a lot - this would be a conflict with Resharper's IntelliSense.
If you use standard IntelliSense in VS, they should work together

**Reply by ayim** [reply]
@Skyppid I'm tracking this in a feedback item here:
https://developercommunity.visualstudio.com/t/GitHub-Copilot-Completions-in-Visual-Stu/10688821

Curious ... on a scale of 1-10, how critical is Resharper's IntelliSense to your overall use of Resharper?
If you were to use Reshaper but with Visual Studio's IntelliSense, how unsatisfied would you be on a scale of 1-10?

--
Aaron Yim
Product Manager for GitHub Copilot in Visual Studio
aayim@microsoft.com

**Reply by Skyppid** [reply]
I'll try around a bit. It's likely that I will continue using Rider as Visual Studio has absolute lackluster support for Source Generators - but if they managed to fix at least the most critical issues I might switch back. I'll try out native IntelliSense then and give some feedback.

**Comment by ayim** [comment]
Thanks! Feel free to reach out to me directly at aayim at microsoft.com with any other feedback related to GitHub Copilot in VS.

--
Aaron Yim
Product Manager for GitHub Copilot in Visual Studio

**Comment by philippdieng** [comment]
I use VS2022 and I really hate and love Copilot there. The automatically shown suggestion are most of the time useless and make me go ragious after some time (My ESC key is suffering sometimes). They are more annyoing than helpful. **If I want a suggestion I will request it. I don't want someone to interfere my thoughts by suggesting useless or invalid code.** In some cases Copilot is really helpful for example /doc or /explain and so on. That's why I love him 🙂

**Reply by waywardcode** [reply]
Now it's more in the way, using copilot for something simple like import suggestions in typescript file ... why? it was working without copilot. Now copilot sometime deletes all lines and just adds the import as the suggestion - so you need to go where imports are and hit enter then it will suggest the correct import, except it ignores your relative path settings -- it's getting be an AI joke.
